qa/sc_evm_native_forger_v2.py

```diff
@@ -889,6 +891,37 @@ def run_test(self):
         current_best_epoch = sc_node_1.block_forgingInfo()["result"]["bestBlockEpochNumber"]
         assert_equal(current_best_epoch, epoch)
 
+        #######################################################################################################
+        # Reward workflow test
+        #######################################################################################################
+
+        ft_pool_amount = 0.5
+        ft_pool_amount_wei = convertZenToWei(ft_pool_amount)
+        forward_transfer_to_sidechain(self.sc_nodes_bootstrap_info.sidechain_id,
+                                      mc_node,
+                                      format_eoa(FORGER_POOL_RECIPIENT_ADDRESS),
+                                      ft_pool_amount,
+                                      mc_return_address=mc_node.getnewaddress(),
+                                      generate_block=False)
+        mc_node.generate(1)
+        generate_next_block(sc_node_1, "second node")
+        # assert Forger Pool balance is updated
+        forger_pool_balance = int(self.sc_nodes[0].rpc_eth_getBalance(format_evm(FORGER_POOL_RECIPIENT_ADDRESS), 'latest')['result'], 16)
+        assert_equal(ft_pool_amount_wei, forger_pool_balance)
+        mc_node.generate(19)
+
+        sc_last_we_block_id = generate_next_block(sc_node_1, "first node")
+
+        self.sc_sync_all()
+
+        # assert Forger Pool balance is distributed
+        forger_pool_balance = int(
+            self.sc_nodes[0].rpc_eth_getBalance(format_evm(FORGER_POOL_RECIPIENT_ADDRESS), 'latest')['result'], 16)
+        assert_equal(0, forger_pool_balance)
+
+        payments = http_block_getFeePayments(sc_node_1, sc_last_we_block_id)['feePayments']
+        print(payments)
```

Comment:
Could you please add some checks on `payments`? It would be nice to have mixed cases as some blocks before fork 1.4, some after, some from forgers with rewardAddress etc. etc.

Reply:
Added some basic checks, but will work on a comprehensive test under https://horizenlabs.atlassian.net/browse/EON-1826 to not block this ticket further